modules/consensus/subscribe.go

```diff
@@ -169,12 +172,17 @@ func (cs *ConsensusSet) managedInitializeSubscribe(subscriber modules.ConsensusS
 		cs.mu.RLock()
 		err = cs.db.View(func(tx *bolt.Tx) error {
 			for i := 0; i < 100 && exists; i++ {
-				cc, err := cs.computeConsensusChange(tx, entry)
-				if err != nil {
-					return err
+				select {
+				case <-cancel:
+					return errors.New("aborting managedInitializeSubscribe")
```

Comment:
I'd like @DavidVorick to chime in here before merging, but in the meantime I think `siasync.ErrStopped` is slightly preferable.